repos/system_upgrade/el7toel8/actors/checkkdegnome/actor.py

```diff
@@ -0,0 +1,167 @@
+import os
+import subprocess
+import time
+
+from leapp.actors import Actor
+from leapp.tags import IPUWorkflowTag, ChecksPhaseTag
+from leapp.reporting import Report, create_report
+from leapp import reporting
+
+_ret_OK = 0  # return value == 0 means everything is OK
+
+
+class CheckKdeGnome(Actor):
+    """
+    Actor will check whether KDE is installed together with GNOME desktop to inform whether we can
+    inhibit the upgrade process. When both are installed, we need to inform the user that KDE will be
+    removed and GNOME will be used instead. If only KDE is installed, we want to inhibit
+    the upgrade process otherwise the user will end up without a desktop.
+    Also if both are installed, but KDE is not used as default desktop, we further check most common
+    KDE apps to inform the user that those will be removed in case the user is using them.
+    """
+    name = 'check_kde_gnome'
+    consumes = ()
+    produces = (Report,)
+    tags = (IPUWorkflowTag, ChecksPhaseTag)
+
+    def is_executable(self, path):
+        """
+        Checks if path exists, if it is file and if is executable.
+        """
+        return os.path.exists(path) and os.path.isfile(path) and os.access(path, os.X_OK)
+
+    def get_xsession(self, path):
+        """
+        Gets current XSession. Return empty string if no XSession found (any reason
+        including bad path etc.)
+        """
+        default_xsession = ""
+        if not (os.path.exists(path) and os.path.isfile(path)):  # Bad path - in container for example
+            return default_xsession
+        with open(path, "r") as f:
+            for line in f.readlines():
+                if "xsession" in line.lower():
+                    default_xsession = line.split("=")[1].lower()
+        return default_xsession
+
+    def check_app_in_use(self, app):
+        """
+        Method return True if application was used in last month, False in other cases.
+        """
+        path = "{0}/.kde/share/config/{1}rc".format(os.environ.get("HOME"), app)
+        if os.path.exists(path) and os.path.isfile(path):
+            last_modified = os.stat(path).st_mtime
+            if last_modified >= int(time.time() - 2592000):
+                self.log.info("Application {0} is actively used".format(app))
```

Comment:
Nit: I'd move the logging out of check methods (have it in the main processing one only) and rewrite as
`return last_modified >= int(time.time() - 2592000)` (just please no magic numbers)